### PR TITLE
Session id with cloudflare

### PR DESCRIPTION
**Pull request by thailyn**
This is a rather janky fix so we can log in while using Session ID.  The user must provide his user agent, and the values of two cloud flare cookies, on the login window.  The first, "CloudFlare ID" (the cookie named `__cfduid`), appears to be pretty static.  However, the second, "CloudFlare PW" (the cookie named `cf_clearance`), appears to expire after about an hour, and it also appears it can not be reused after a failed login attempt.  The user must navigate to `https://www.pathofexile.com/login` in his browser to force refreshing the `cf_clearance` cookie.  I haven't tried keeping Procurement open for a while and seeing if that cookie invalidates after a while; the user might end up needing to re-log-in every so often.

I personally think the additions to the UI are awful, and would love for them to be redone by a more loving hand, but... it works.

Partially addresses #901.

### CODE REVIEW

**Review by aydjay — CHANGES_REQUESTED** [review]
Fix works, small amendments required but we should absolutely expedite this into a release to restore Procurement to a working state.

File: Procurement/ViewModel/LoginWindowViewModel.cs [hunk 1]
```diff
@@ -77,6 +77,42 @@ public bool ForceRefresh
...
 
+        private string _userAgent;
+        public string UserAgent
+        {
+            get { return _userAgent; }
+            set
+            {
+                _userAgent = value;
+                Settings.UserSettings["UserAgent"] = value;
+                OnPropertyChanged("UserAgent");
```
[the remark this answers is not in the document]
Comment (thailyn) [reply Procurement/ViewModel/LoginWindowViewModel.cs]: Yeah, the new base class is probably a good idea, but outside the cope of this PR.

File: POEApi.Transport/HttpTransport.cs [hunk 2]
```diff
@@ -42,10 +42,24 @@ protected enum HttpMethod { GET, POST }
 
         private static TaskThrottle taskThrottle = new TaskThrottle(TimeSpan.FromMinutes(1), 42, 42);
```
[the remark this answers is not in the document]
Comment (thailyn) [reply POEApi.Transport/HttpTransport.cs]: Done.

File: Procurement/View/LoginView.xaml [hunk 3]
```diff
@@ -28,25 +28,71 @@
...
+                <Label x:Name="lblCloudFlareId" Content="CloudFlare ID" Grid.Row="4" Grid.Column="0"
+                       FontFamily="../Resources/#Fontin" Foreground="#FFAB9066" FontSize="14" VerticalAlignment="Top" >
+                </Label>
+                <Label x:Name="lblCloudFlareClearance" Content="CloudFlare PW" Grid.Row="5" Grid.Column="0"
+                       FontFamily="../Resources/#Fontin" Foreground="#FFAB9066" FontSize="14" VerticalAlignment="Top" >
+                </Label>
 
                 <TextBox Name="txtLogin" Text="{Binding Email}" Grid.Row="1" Grid.Column="1" Height="22" Width="330" HorizontalAlignment="Left" VerticalAlignment="Top" Background="Black" BorderBrush="#FF76591B" Foreground="#FFAB9066" SelectionBrush="#FFCABE9F"></TextBox>
                 <PasswordBox Name="txtPassword" Grid.Row="2" Grid.Column="1" Height="22" Width="330" Background="Black" BorderBrush="#FF76591B" HorizontalAlignment="Left" VerticalAlignment="Top"  Foreground="#FFAB9066" SelectionBrush="#FFCABE9F"/>
+                <TextBox Name="txtUserAgent" Text="{Binding UserAgent}" Grid.Row="3" Grid.Column="1"
```
[the remark this answers is not in the document]
Comment (thailyn) [reply Procurement/View/LoginView.xaml]: Done.

File: POEApi.Transport/CachedTransport.cs [hunk 4]
```diff
@@ -17,6 +17,19 @@ public class CachedTransport : ITransport
 
         public event ThottledEventHandler Throttled;
```
[the remark this answers is not in the document]
Comment (thailyn) [reply POEApi.Transport/CachedTransport.cs]: Got it.  Initially I had this property not forwarding the user agent, which was wrong, and I forgot to remove the backing variable after making that switch.

**Comment by aydjay** [comment]
This can be closed with #903's merging

**Pull request closed without merging**